plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheckTask.kt

```diff
@@ -72,6 +73,9 @@ open class KtlintCheckTask @Inject constructor(
             }
             .filter { it.enabled.get() }
 
+    @get:Nested
+    internal val filterAction: Property<Action<PatternFilterable>> = objectFactory.property()
```

Comment:
🤔 I.... Hrmmm...
I don't think this should be a field, I think it should be applied by the plugin logic before the `@TaskAction` is evaluated.

If the `filterAction`changes, the `source` will change thus invalidating the task.

Reply:
The problem is that `filterAction` is not available from configuration(returns default value), when task is created. Any clues how to get it on task creation time?

Reply:
Ooooh, interesting question. I'm thinking about it. Let me get back to you.